atlp-rwanda/space-bn-frontend
# #176038071 ft(hotel): implement list of hotels page

#### What does this PR do?

- Implement hotel listing page

#### Description of Task to be completed? 

- Display list of hotels

- Add search bar

- Add search functionalities

- Add map along side the hotels' list

- Add infinite scroll

- Make map interactive

- Link page to hotel full details page

#### How should this be manually tested?

- Clone the repo 

- In your home directory  cd this branch **`ft-hotels-page-#176038071`**

- Then run **`npm install`** to install all dependencies

- Run **`npm start`** to start the server

- Run **`npm test`** to run tests

- In your default browser navigate to **`http://localhost:3000/hotel`**

> You should see a hotel listing page


#### Any background context you want to provide?

- To see the map you should have provided the credentials in the **`.env`**

- **Credentials should be provided like this**

```
REACT_APP_SPACE_BN_TOKEN=YourToken

REACT_APP_SECRET_USERNAME=authenticatedUsername

REACT_APP_SECRET_PASSWORD=authenticatePassword

```

- Check credentials values on **slack channel**

#### What are the relevant Pivotal Tracker stories?

- [#176038071](https://www.pivotaltracker.com/n/projects/2476677/stories/176038071)

#### Screenshots (if appropriate)
<img width="1440" alt="Screen Shot 2020-12-14 at 06 42 12" src="https://user-images.githubusercontent.com/54619678/102041263-aaef3b00-3dd7-11eb-8162-b380f00bd9b4.png">



#### Questions:

- N/A
